### ISSUE TITLE
TimeZonePicker: Make accessible to screen reader

### ISSUE THREAD

**Issue by tskarhed** [grafana/grafana#36915]
**Steps to reproduce**:
1. Click change time range
2. Click change time zone
3. Listen to option in the Select

**Actual Result**:
It doesn't read the options, only some letters from it. Tested with Orca.

**Expected Result**
For it to read something somewhat comprehensive.

**Environment**:
- Grafana version:
- Data source type & version:
- User OS & Browser:
- Others:

**Comment by tharun208** [comment]
@tskarhed I like to work on this

**Comment by tskarhed** [comment]
This one might turn out to be very complex quite quickly. Hopefully there are no needed changes to the underlying `Select` component.

A qualified guess is that `TimeZoneOption` and `TimeZoneGroup` needs better markup, possibly new labels.

**Comment by tharun208** [comment]
@tskarhed, I think you already working on PR - https://github.com/grafana/grafana/pull/36912. Should this above issue needs a fix as a separate PR or can be included with your changes?

**Comment by thisisobate** [comment]
I tested `Timezonepicker` using VoiceOver app on MacOS, and screen reader chrome extension. They all voice out the correct options. Would proceed to close this since I couldn't reproduce and also mark this task as completed in the [a11y phase 1](https://github.com/grafana/grafana/issues/35613) backlog.
